CRM/EventsExtras/Hook/BuildForm/EventFee.php

```diff
@@ -39,7 +40,9 @@ private function setDefaults(&$form){
      $settings = [$paymentProcessor, $paymentProcessorDefault];
      $settingValues = SettingsManager::getSettingsValue($settings);
      if ($settingValues[$paymentProcessor] == 0){
-       $defaults['default_role_id'] = $settingValues[$paymentProcessorDefault];
+      $defaultSettingString = implode(CRM_Core_DAO::VALUE_SEPARATOR, $settingValues[$paymentProcessorDefault]);
+      $paymentProcessorDefaultValue = (array_fill_keys(explode(CRM_Core_DAO::VALUE_SEPARATOR, $defaultSettingString), '1'));
```

Comment:
Not sure why we have to implode and explode, Can't the keys be filled directly. i.e `array_fill_keys($settingValues[$paymentProcessorDefault], 1)` ?

Reply:
No we can't. They way html form capture data and they way civicrm store database is different. We should store data (checkboxes) as per civicrm.